### PR TITLE
feat(ctb): Minimum large preimage proposal size in `PreimageOracle`

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Adds a minimum large preimage proposal size in the `PreimageOracle`, and makes the min size + challenge period configurable. Without a minimum size, it would be low-cost for attackers to spam large preimage proposals that the `op-challenger` must counter. With a minimum size, we encourage use of the regular path for smaller preimages and add a natural spam-deterrent (cost of proposing a large preimage >= ~2MB far outweighs the cost of challenging.)

**Metadata**
closes https://github.com/ethereum-optimism/client-pod/issues/468

### CODE REVIEW

**Comment by clabby** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #9035** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9035?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #9039** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9039?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
      * **PR #9052** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9052?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9039?utm_source=stack-comment).

File: packages/contracts-bedrock/src/cannon/PreimageOracle.sol [hunk 1]
```diff
@@ -13,15 +13,17 @@ import "src/cannon/libraries/CannonTypes.sol";
...
 contract PreimageOracle is IPreimageOracle {
     ////////////////////////////////////////////////////////////////
-    //                         Constants                          //
+    //                   Constants & Immutables                   //
     ////////////////////////////////////////////////////////////////
 
+    /// @notice The duration of the large preimage proposal challenge period.
+    uint256 public immutable CHALLENGE_PERIOD;
+    /// @notice The minimum size of a preimage that can be proposed in the large preimage path.
+    uint256 public immutable MIN_LPP_SIZE_BYTES;
```
Comment (Inphi) [inline packages/contracts-bedrock/src/cannon/PreimageOracle.sol]: per style, immutables should be internal with manual getters.

File: packages/contracts-bedrock/src/cannon/PreimageOracle.sol [hunk 2]
```diff
@@ -195,6 +200,9 @@ contract PreimageOracle is IPreimageOracle {
         // The part offset must be within the bounds of the claimed size + 8.
         if (_partOffset >= _claimedSize + 8) revert PartOffsetOOB();
 
+        // The claimed size must be at least 1.8MB.
```
Comment (Inphi) [inline packages/contracts-bedrock/src/cannon/PreimageOracle.sol]: maybe qualify that this is typically the case for mainnet/testnets or something.

**Review by Inphi — APPROVED** [review]

File: packages/contracts-bedrock/src/cannon/PreimageOracle.sol (hunk 2 above)
Comment (clabby) [reply packages/contracts-bedrock/src/cannon/PreimageOracle.sol]: Ah whoops - Forgot to modify this comment, good catch.

**Pull request merged**